worker-markup-core/src/main/java/com/github/cafdataprocessing/worker/markup/core/MarkupDocument.java

```diff
@@ -0,0 +1,173 @@
+import com.github.cafdataprocessing.worker.markup.core.exceptions.MarkupWorkerExceptions;
+import com.google.common.collect.Multimap;
+import com.hpe.caf.api.Codec;
+import com.hpe.caf.api.ConfigurationException;
+import com.hpe.caf.api.ConfigurationSource;
+import com.hpe.caf.api.worker.DataStore;
+import com.hpe.caf.api.worker.DataStoreSource;
+import com.hpe.caf.codec.JsonCodec;
+import com.hpe.caf.util.ref.DataSource;
+import com.hpe.caf.util.ref.ReferencedData;
+import com.hpe.caf.worker.document.model.Document;
+import com.hpe.caf.worker.markup.HashConfiguration;
+import com.hpe.caf.worker.markup.MarkupWorkerResult;
+import com.hpe.caf.worker.markup.MarkupWorkerStatus;
+import com.hpe.caf.worker.markup.MarkupWorkerTask;
+import com.hpe.caf.worker.markup.OutputField;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import org.jdom2.JDOMException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MarkupDocument
+{
+    private static final Logger LOG = LoggerFactory.getLogger(MarkupDocument.class);
+
+    final ExecutorService jepThreadPool = Executors.newSingleThreadExecutor();
+    final EmailSplitter emailSplitter = new EmailSplitter(new JepExecutor(jepThreadPool));
+
+    /**
+     * Public method to expose Markup worker logic to a document worker
+     *
+     * @param document Instance of document worker document object
+     * @param hashConfiguration The hash configuration to use when generating document hashes
+     * @param outputFields The list of fields the worker should output
+     * @param isEmail if the document is of email type.
+     * @throws InterruptedException throws in cases of a thread being interrupted during processing.
+     * @throws com.hpe.caf.api.ConfigurationException throws when configuration for worker is malformed or missing.
+     * @throws org.jdom2.JDOMException throws when an error occurs during parsing.
+     * @throws java.util.concurrent.ExecutionException throws when an error occurs during email splitting.
+     */
+    public void markupDocument(final Document document, final List<HashConfiguration> hashConfiguration,
+                               final List<OutputField> outputFields, final boolean isEmail)
+        throws InterruptedException, ConfigurationException, JDOMException, ExecutionException
+    {
+        final MarkupWorkerConfiguration config = document.getApplication().getService(ConfigurationSource.class)
+            .getConfiguration(MarkupWorkerConfiguration.class);
+        final DataStore dataStore = document.getApplication().getService(DataStore.class);
+
+        try {
+            MarkupWorkerResult result = markupDocument(ConvertSourceData.getSourceData(document), hashConfiguration, outputFields, isEmail,
+                                                       new JsonCodec(), dataStore, config);
```

Comment:
You shouldn't hardcode JsonCodec here. It's configurable, right? I added service locator registration for the codec on the Application object (in the Document Worker framework) so you should be able to request it like the data store.